http/logical.go

```diff
@@ -160,8 +160,8 @@ func respondLogical(w http.ResponseWriter, r *http.Request, req *logical.Request
 		}
 
 		// Check if this is a raw response
-		if _, ok := resp.Data[logical.HTTPContentType]; ok {
-			respondRaw(w, r, req.Path, resp)
+		if _, ok := resp.Data[logical.HTTPStatusCode]; ok {
```

Comment:
PKI always sets all three. Generally speaking it's better to key off a status code since a raw responder should always be setting a status code but Go will attempt to determine content type if it's not set, so there's a fallback.